scipy/stats/tests/test_hypotests.py

```diff
@@ -1236,3 +1236,211 @@ def test_same_input(self):
+        ['2', '3', '4.34', '1.561605075444930', '7.118394924555070', '1'],
+        ['2', '1', '6.08', '2.740784878675930', '9.419215121324070', '1'],
+        ['3', '2', '-4.34', '-7.11839492455507', '-1.56160507544493', '1'],
+        ['3', '1', '1.74', '-1.96452656607342', '5.444526566073420', '0'],
+        ['1', '2', '-6.08', '-9.41921512132407', '-2.74078487867593', '1'],
+        ['1', '3', '-1.74', '-5.44452656607343', '1.964526566073420', '0']
+    ]
+
+    @pytest.mark.parametrize("data,SAS_res", ((data_same_size, sas_same_size),
+                                              (data_diff_size, sas_diff_size),
+                                              (extreme_size, sas_extreme)),
+                             ids=["equal size sample", "unequal sample size",
+                                  "extreme sample size differences"])
+    def test_compare_sas(self, data, SAS_res):
+        '''
+        SAS code used to generate results for each sample:
+        DATA ACHE;
+        INPUT BRAND RELIEF;
+        CARDS;
+        1 24.5
+        1 23.5
+        1 26.28
+        1 26.4
+        1 27.1
+        1 29.9
+        1 30.1
+        1 30.1
+        2 28.4
+        2 34.2
+        2 29.5
+        2 32.2
+        2 30.1
+        3 26.1
+        3 28.3
+        3 24.3
+        3 26.2
+        3 27.8
+        ;
+        ods graphics on;   ODS RTF;ODS LISTING CLOSE;
+           PROC ANOVA DATA=ACHE;
+           CLASS BRAND;
+           MODEL RELIEF=BRAND;
+           MEANS BRAND/TUKEY CLDIFF;
+           TITLE 'COMPARE RELIEF ACROSS MEDICINES  - ANOVA EXAMPLE';
+           ods output  CLDiffs =tc;
+        proc print data=tc;
+            format LowerCL 17.16 UpperCL 17.16 Difference 17.16;
+            title "Output with many digits";
+        RUN;
+        QUIT;
+        ODS RTF close;
+        ODS LISTING;
+        # The standard output, not showing the additional digits. The code
+        # example above outputs a table with more digits.
+        Brand Comparison    Mean Difference  Simultanious 95% Confidence Limits
+        2 - 1               3.645               0.268           7.022   1
+        2 - 3               4.340               0.593           8.087   1
+        1 - 2              -3.645              -7.022          -0.268   1
+        1 - 3               0.695              -2.682           4.072   0
+        3 - 2              -4.340              -8.087          -0.593   1
+        3 - 1              -0.695              -4.072           2.682   0
+        '''
+
+        res = stats.tukey_hsd(*data)
+        conf = res.confidence_interval()
+        for comp in SAS_res:
+            i, j, stat, lower, upper, sig = np.asarray(comp, dtype=float)
+            # make int for slicing, zero index
+            i, j = int(i) - 1, int(j) - 1
+            assert_allclose(stat, res.statistic[i, j], atol=1e-13)
+            assert_allclose(lower, conf.low[i, j], atol=1e-4)
+            assert_allclose(upper, conf.high[i, j], atol=1e-4)
```

Comment:
Continuation of:
> Now that we have more digits, can we make the tolerance tighter?

You wrote:
> Unfortunately not for the confidence levels. The statistic is accurate to 13 digits, though.

This is not really impressive because the sample means have only two decimal places, so I would expect perfect agreement.

Even after the correction to the number of degrees of freedom this is the best we can do?

Reply:
I'm not really sure yet why it doesn't match. I also figured out how to get tukey's hsd in minitab, and the minitab confidence interval output is only accurate to 2 - 3 digits for **both SAS and this implementation.** I would have thought that there would be good agreement between at least one set of these.

Reply:
Hmm. Could be their implementations of the studentized range distribution. Maybe try https://www.mathworks.com/help/stats/multcompare.html?

Reply:
I assume so. Matlab confidence intervals are good to 8 digits of precision. I will add a test for it.

Reply:
Excellent!